src/overcooked_ai_py/agents/benchmarking.py

```diff
@@ -20,83 +19,65 @@ class AgentEvaluator(object):
 
-    def evaluate_one_optimal_one_random(self, num_games, display=True):
-        a0 = CoupledPlanningAgent(self.mlp)
-        a1 = RandomAgent()
+    def evaluate_human_model_pair(self, num_games=1, display=False, native_eval=False):
+        a0 = GreedyHumanModel(self.env.mlam)
+        a1 = GreedyHumanModel(self.env.mlam)
         agent_pair = AgentPair(a0, a1)
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
-
-    def evaluate_one_optimal_one_greedy_human(self, num_games, h_idx=0, display=True):
-        h = GreedyHumanModel(self.mlp)
-        r = CoupledPlanningAgent(self.mlp)
-        agent_pair = AgentPair(h, r) if h_idx == 0 else AgentPair(r, h)
-        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display)
-
-    def evaluate_agent_pair(self, agent_pair, num_games, game_length=None, start_state_fn=None, metadata_fn=None, metadata_info_fn=None, display=False, info=True):
-        horizon_env = self.env.copy()
-        horizon_env.horizon = self.env.horizon if game_length is None else game_length
-        horizon_env.start_state_fn = self.env.start_state_fn if start_state_fn is None else start_state_fn
-        horizon_env.reset()
-        return horizon_env.get_rollouts(agent_pair, num_games=num_games, display=display, info=info, metadata_fn=metadata_fn, metadata_info_fn=metadata_info_fn)
+        return self.evaluate_agent_pair(agent_pair, num_games=num_games, display=display, native_eval=native_eval)
+
+    def evaluate_agent_pair(self, agent_pair, num_games, game_length=None, start_state_fn=None, metadata_fn=None, metadata_info_fn=None, display=False, dir=None, info=True, native_eval=False):
+        # this index has to be 0 because the Agent_Evaluator only has 1 env initiated
+        # if you would like to evaluate on a different env using rllib, please modifiy
+        # rllib/ -> rllib.py -> get_rllib_eval_function -> _evaluate
+
+        # native eval: using self.env in evaluation instead of creating a copy
+        # this is particulally helpful with variable MDP, where we want to make sure
+        # the mdp used in evaluation is the same as the native self.env.mdp
+        if native_eval:
+            return self.env.get_rollouts(agent_pair, num_games=num_games, display=display, dir=dir, info=info,
+                                         metadata_fn=metadata_fn, metadata_info_fn=metadata_info_fn)
+        else:
+            horizon_env = self.env.copy()
+            horizon_env.horizon = self.env.horizon if game_length is None else game_length
+            horizon_env.start_state_fn = self.env.start_state_fn if start_state_fn is None else start_state_fn
+            horizon_env.reset()
+            return horizon_env.get_rollouts(agent_pair, num_games=num_games, display=display, dir=dir, info=info,
+                                            metadata_fn=metadata_fn, metadata_info_fn=metadata_info_fn)
```

Comment:
I'm not entirely sure in what cases doing the additional steps of copying the environment and resetting it should cause things to break for multi-layout evaluation? In fact, I feel like we probably want to build the multi-layout environment to work around this constraint, that seems quite intuitive: "If i have an environment, I copy it and reset it, it should still be the same environment." To add support for the case of environments with multiple MDPs, I'm guessing some additional optional params might need to be added to the `reset` function – in fact, you have already added `regen_mdp` (or something along those lines), that seems like a good idea. In fact, if `regen_mdp=False`, I'm guessing that this function should work as you intended (that might just require adding `regen_mdp` to the signature of `evaluate_agent_pair`)? Not sure what exact use case you had in mind?

Another thing that comes to mind as a useful signature addition to `.reset()` is `schedule_loc` (a 0 to 1 float indicating how far in we are in the schedule, that is also passed into the mdp generation fn) – as I was alluding to in some other comments.

In any case, this code could be simplified to the following, in order to have less repeats:
```suggestion
    def evaluate_agent_pair(self, agent_pair, num_games, game_length=None, start_state_fn=None, metadata_fn=None, metadata_info_fn=None, display=False, dir=None, info=True, native_eval=False):
        # this index has to be 0 because the Agent_Evaluator only has 1 env initiated
        # if you would like to evaluate on a different env using rllib, please modifiy
        # rllib/ -> rllib.py -> get_rllib_eval_function -> _evaluate

        # native eval: using self.env in evaluation instead of creating a copy
        # this is particulally helpful with variable MDP, where we want to make sure
        # the mdp used in evaluation is the same as the native self.env.mdp
        if not native_eval:
            horizon_env = self.env.copy()
            horizon_env.horizon = self.env.horizon if game_length is None else game_length
            horizon_env.start_state_fn = self.env.start_state_fn if start_state_fn is None else start_state_fn
            horizon_env.reset()
        return self.env.get_rollouts(agent_pair, num_games=num_games, display=display, dir=dir, info=info,
                                         metadata_fn=metadata_fn, metadata_info_fn=metadata_info_fn)
```

Reply:
directly copying and regenerating will be a problem, since "copying" the OvercookedEnv only copies the generator function, not the mdp inside it. Copy the mdp inside it won't be good enough, because the reset in the env is going to generate a new mdp anyway. (Basically, copying over the lottery box (env) doesn't mean that the same guy will end up winning the lottery (env.mdp).)

Regarding passing in additional parameter: I have left space in the code to be able to do this. But I feel like it really should be in a different pull request, where we have a better idea about how we want to do MDP parameter generation.

Reply:
This scheduling functionality and tests has been added on the newest commit b16c69b8b389cb38342ac70d93ebbdfe5368520a

Regarding the evaluate_agent_pair, I don't think the two are necessarily equivalent, because one is running from `self.env` and the other one is running on the cloned `horizon_env`. We need to have two different return statements, otherwise it will not produce the desired behavior.

Reply:
horizon_env is usually equivalent to self.env (unless horizon or start state parameters are overwritten). But yeah, I agree that for now we can keep it this way!

Reply:
Sounds good